File: Java/CSharpParserBase.java
import org.antlr.v4.runtime.*;
import java.util.Stack;

public abstract class CSharpParserBase extends Parser
{
    protected CSharpParserBase(TokenStream input)
    {
	super(input);
    }

    public boolean P1()
    {
        return (((CSharpParser.Right_arrowContext)this._ctx).first.getTokenIndex() + 1) == (((CSharpParser.Right_arrowContext)this._ctx).second.getTokenIndex());
    }

    public boolean P2()
    {
        return (((CSharpParser.Right_shiftContext)this._ctx).first.getTokenIndex() + 1) == (((CSharpParser.Right_shiftContext)this._ctx).second.getTokenIndex());
    }

    public boolean P3()
    {
        return (((CSharpParser.Right_shift_assignmentContext)this._ctx).first.getTokenIndex() + 1) == (((CSharpParser.Right_shift_assignmentContext)this._ctx).second.getTokenIndex());
    }
}

File: Go/csharp_parser_base.go
package parser

import (
	"github.com/antlr/antlr4/runtime/Go/antlr"
)

// CSharpParserBase implementation.
type CSharpParserBase struct {
	*antlr.BaseParser
}

func (p *CSharpParser) P1() bool {
	x := p.CSharpParserBase
	y := x.BaseParser
	z := y.GetParserRuleContext()
	d := z.(IRight_arrowContext)
	return (d.GetFirst().GetTokenIndex() + 1) == (d.GetSecond().GetTokenIndex())
}

func (p *CSharpParser) P2() bool {
	x := p.CSharpParserBase
	y := x.BaseParser
	z := y.GetParserRuleContext()
	d := z.(IRight_shiftContext)
	return (d.GetFirst().GetTokenIndex() + 1) == (d.GetSecond().GetTokenIndex())
}

func (p *CSharpParser) P3() bool {
	x := p.CSharpParserBase
	y := x.BaseParser
	z := y.GetParserRuleContext()
	d := z.(IRight_shift_assignmentContext)
	return (d.GetFirst().GetTokenIndex() + 1) == (d.GetSecond().GetTokenIndex())
}
